### DISCUSSION TITLE
`storeToRefs` and `reactive()` in setup store

### DISCUSSION THREAD

**Discussion by Tim-OuY** [vuejs/pinia#2319] [Help and Questions] [answered]
Code Example: https://stackblitz.com/edit/vue-pinia-store-state-reactivity-7wgfwu?file=src%2FApp.vue

Directly change `state.properties` with `useStore()` instance will break reactivity of `storeToRefs()`, However, objects defined by `reactive()` can actually change directly and still keep the reactivity (eg. `reactiveArray` in this [case](https://stackblitz.com/edit/vue-pinia-store-state-reactivity-7wgfwu?file=src%2FApp.vue)).

Is it by design or is there a bug?

**Comment by posva** [comment, marked as answer]
This is by design as reactive [objects cannot be replaced](https://vuejs.org/guide/essentials/reactivity-fundamentals.html#limitations-of-reactive)
